set.mm

```diff
@@ -121414,6 +121414,20 @@ subset of complex numbers ( ~ nnsscn ), in contrast to the more elementary
       ( cn wcel cmul co nnmulcl mp2an ) AEFBEFABGHEFCDABIJ $.
   $}
 
+  $( "Minus times minus is plus, The reason for this we need not discuss."  (W.
+     H. Auden, as quoted in M. Guillen "Bridges to Infinity", p. 64, see also
+     Metamath Book, p. 5) This statement, formalized to "The product of two
+     negative integers is a positive integer", is proved by the following
+     theorem, therefore it actually needs not be discussed anymore.  "The
```

Comment:
needs --> need

Also: to nitpick, what you proved is "the product of the opposites of two positive integers is a positive integer". A formalization more faithful to the first sentence would be
```
|- ( ( -u A e. NN /\ -u B e. NN ) -> ( A x. B ) e. NN )
```
But that's not really a problem and I'm fine with your statement (though maybe you could add the above remark in the comment).

Reply:
> needs --> need

OK, is already corrected.

> Also: to nitpick, what you proved is "the product of the opposites of two positive integers is a positive integer". A formalization more faithful to the first sentence would be
> 
> ```
> |- ( ( -u A e. NN /\ -u B e. NN ) -> ( A x. B ) e. NN )
> ```
> 
> But that's not really a problem and I'm fine with your statement (though maybe you could add the above remark in the comment).

The opposite of a positive integer is a negative integer! Furthermore, the rule "minus times minus is plus" is often presented with explicit integers in the form "-3 x -4 = 12" (especially in school), and that fits to ~ nnmtmip. And finally, since there is no reverse closure of -u, we cannot derive A e. ZZ from -u A e. NN, and therefore the proposed theorem cannot be proven.
To be precise, one would have to define the set of negative integers ZZ- = { z e. ZZ | z < 0 }. Then 

|- ( ( A e. ZZ- /\ B e. ZZ- ) -> ( A x. B ) e. NN )

would be the appropriate theorem.

To make thinks not too complicated, I would like to leave the theorem as it is.

Reply:
You're right, the statement should be
```
|- ( ( ( A e. ZZ /\ -u A e. NN ) /\ ( B e. ZZ /\ -u B e. NN ) ) -> ( A x. B ) e. NN )
```
And indeed, it assumes that "a negative integer is an integer whose opposite is a positive integer".
As for deciding which is the better definition:
```
A negative integer is the opposite of a positive integer.
A negative integer is an integer whose opposite is a positive integer.
```
maybe the first version is better indeed !